packages/xo-server/src/api/test.js

```diff
@@ -86,3 +89,37 @@ copyVm.resolve = {
   vm: ['vm', 'VM'],
   sr: ['sr', 'SR'],
 }
+
+// -------------------------------------------------------------------
+
+export async function changeConnectedXapiHostname({
+  hostname,
+  newObject,
+  oldObject,
+}) {
+  const xapi = this.getXapi(oldObject)
+  const { pool: currentPool } = xapi
+
+  xapi._setUrl({ ...xapi._url, hostname })
+  await fromEvent(xapi.objects, 'finish')
+  if (xapi.pool.$id === currentPool.$id) {
+    await fromEvent(xapi.objects, 'finish')
+  }
+
+  assert(xapi.pool.$id !== currentPool.$id)
+  assert.doesNotThrow(() => this.getXapi(newObject))
+  assert.throws(() => this.getXapi(oldObject))
+}
+
+changeConnectedXapiHostname.description =
+  'change the connected XAPI hostname and check if the pool and the local cache are updated'
+
+changeConnectedXapiHostname.permission = 'admin'
+
+changeConnectedXapiHostname.params = {
+  hostname: { type: 'string' },
+  // new connection's XO object
```

Comment:
add a `description` property to the object.

Reply:
It's not taken account by the` xo-cli`.

Reply:
Neither are the comments.

But the `description` is standard in JSON schema and could one day be displayed.